### PR TITLE
build: Apache groovy 4.0.9

### PR DESCRIPTION
**Pull request by sdelamo**

### CODE REVIEW

File: inject-groovy/src/main/groovy/io/micronaut/ast/groovy/visitor/LoadedVisitor.groovy [hunk 1]
```diff
@@ -94,6 +95,7 @@ class LoadedVisitor implements Ordered {
         return true
     }
 
+    @CompileDynamic
```
Comment (sdelamo) [inline inject-groovy/src/main/groovy/io/micronaut/ast/groovy/visitor/LoadedVisitor.groovy:98]: without `@CompilieDynamic`

```
 [Static type checking] - Non-static method java.lang.Object#hashCode cannot be called from static context
 @ line 99, column 16.
           return visitor.getClass().hashCode()
```

Comment (graemerocher) [reply inject-groovy/src/main/groovy/io/micronaut/ast/groovy/visitor/LoadedVisitor.groovy:98]: Seems like a regression in Groovy

**Review by graemerocher — APPROVED** [review]

File: inject-groovy/src/main/groovy/io/micronaut/ast/groovy/visitor/LoadedVisitor.groovy (hunk 1 above)
Comment (sdelamo) [reply inject-groovy/src/main/groovy/io/micronaut/ast/groovy/visitor/LoadedVisitor.groovy:98]: it is present since 4.0.7

Comment (sdelamo) [reply inject-groovy/src/main/groovy/io/micronaut/ast/groovy/visitor/LoadedVisitor.groovy:98]: I submitted an issue https://issues.apache.org/jira/browse/GROOVY-10939

**Pull request merged**